tests/key_field/key_field_sample_generated.h

```diff
@@ -153,63 +123,290 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Bar FLATBUFFERS_FINAL_CLASS {
+    return flatbuffers::EndianScalar(tag_);
+  }
+};
+FLATBUFFERS_STRUCT_END(Color, 16);
 
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Apple FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint8_t tag_;
+  int8_t padding0__;  int16_t padding1__;
+  keyfield::sample::Color color_;
 
-struct FooTableT : public flatbuffers::NativeTable {
-  typedef FooTable TableType;
-  int32_t a = 0;
-  int32_t b = 0;
-  std::string c{};
-  std::vector<keyfield::sample::Baz> d{};
-  std::vector<keyfield::sample::Bar> e{};
+ public:
+  Apple()
+      : tag_(0),
+        padding0__(0),
+        padding1__(0),
+        color_() {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Apple(uint8_t _tag, const keyfield::sample::Color &_color)
+      : tag_(flatbuffers::EndianScalar(_tag)),
+        padding0__(0),
+        padding1__(0),
+        color_(_color) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  uint8_t tag() const {
+    return flatbuffers::EndianScalar(tag_);
+  }
+  const keyfield::sample::Color &color() const {
+    return color_;
+  }
+  bool KeyCompareLessThan(const Apple * const o) const {
+    return KeyCompareWithValue(o->color()) < 0;
+  }
+  int KeyCompareWithValue(const keyfield::sample::Color  &_color) const {
+    const auto lhs_color = color();
+    const auto rhs_color = _color;
+    const auto lhs_color_rgb = lhs_color.rgb();
+    const auto rhs_color_rgb = rhs_color.rgb();
+    for (flatbuffers::uoffset_t i = 0; i < lhs_color_rgb->size(); i++) {
+      const auto lhs_color_rgb_elem = lhs_color_rgb->Get(i);
+      const auto rhs_color_rgb_elem = rhs_color_rgb->Get(i);
+      if (lhs_color_rgb_elem != rhs_color_rgb_elem)
+        return static_cast<int>(lhs_color_rgb_elem > rhs_color_rgb_elem) - static_cast<int>(lhs_color_rgb_elem < rhs_color_rgb_elem);
+    }
+    const auto lhs_color_tag = lhs_color.tag();
+    const auto rhs_color_tag = rhs_color.tag();
+    if (lhs_color_tag !=  rhs_color_tag)
+      return static_cast<int>(lhs_color_tag > rhs_color_tag) - static_cast<int>(lhs_color_tag < rhs_color_tag);
+    return 0;
+  }
 };
+FLATBUFFERS_STRUCT_END(Apple, 20);
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Fruit FLATBUFFERS_FINAL_CLASS {
+ private:
+  keyfield::sample::Apple a_;
+  uint8_t b_;
+  int8_t padding0__;  int16_t padding1__;
+
+ public:
+  Fruit()
+      : a_(),
+        b_(0),
+        padding0__(0),
+        padding1__(0) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Fruit(const keyfield::sample::Apple &_a, uint8_t _b)
+      : a_(_a),
+        b_(flatbuffers::EndianScalar(_b)),
+        padding0__(0),
+        padding1__(0) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  const keyfield::sample::Apple &a() const {
+    return a_;
+  }
+  bool KeyCompareLessThan(const Fruit * const o) const {
+    return KeyCompareWithValue(o->a()) < 0;
+  }
+  int KeyCompareWithValue(const keyfield::sample::Apple  &_a) const {
+    const auto lhs_a = a();
+    const auto rhs_a = _a;
+    const auto lhs_a_tag = lhs_a.tag();
```

Comment:
As Apples key is Color, do we need to check tag too?

Reply:
And, can we just call KeyCompareWithValue of Apple ?

Reply:
In this case, apple is the key of fruit. That's why all the fields in apple needs to be checked. Also when you say "call KeyCompareWithValue of Apple", do you mean have it like this way`  int KeyCompareWithValue(const Apple  &_a) const ` ?

Reply:
Apple is the key of fruit, so we want to compare apples, As apples also have key which is Color, For comparing apples, should we ignore Apple keys and compare whole content or comparing its key is just enough?

And for second part, I mean, `lhs_a.KeyCompareWithValue(rhs_a->color);`

Reply:
I'm not sure about using nested keys instead of the whole content, I think that complicate this logic and might be confusing. 
For the second part, I think I can reuse KeyCompareWithValue, let me check and change the generator.